luarules/gadgets/unit_water_depth_damage.lua

```diff
@@ -0,0 +1,179 @@
+function gadget:GetInfo()
+	return {
+		name = "Water Crush and Collision Damage",
+		desc = "Creates and handles water collision events, and kills units stuck underwater",
+		author = "SethDGamre",
+		date = "2024.9.22",
+		license = "GNU GPL, v2 or later",
+		layer = 0,
+		enabled = true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+--use customParams.water_fall_damage_multiplier = 1.0 to change the amount of fall damage taken by specific units.
+
+--any maxWaterDepth movedef equal to or above this number will not take drowning damage.
+local isDrownableMaxWaterDepth = 5000
```

Comment:
ideally units would just start taking damage when deeper than their maxdepth instead of this magic immunity threshold

Reply:
The immunity threshold is for performance reasons. If the unit is amphibious, hovercraft, or a ship then it will always be in the water and would slow down performance by being checked for depth all the time that will never be deep enough to take damage. 

I tried this approach before. Turns out units can achieve depth greater than their max depth threshold based on how steep the slope is that they are on at the time. For example, a grunt has a max depth of -22 but I was able to get it down to -31.

I was unable to recreate the results using trigonometry and groundnormals so the current approach of testing the ground position was used instead and seems to work fine with less code. 

```suggestion
local isDrownableMaxWaterDepth = 5000
```

Reply:
```suggestion
-- performance optimisation to avoid checking amphs and hovers
local isDrownableMaxWaterDepth = 5000
```